### PR TITLE
Update Pull Request template based on eng team discussion

### PR DESCRIPTION
**Pull request by dkatzz**
### Description

This includes data migration updates and more clarity around the release notes section

https://docs.google.com/spreadsheets/d/1jXuOWl9TxssHh3A1AkmKi2gva9LRUhE0PQStJa-4yWs/edit?gid=586038575#gid=586038575

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -26,6 +27,14 @@ Read the guidelines [here](https://github.com/civiform/civiform/wiki/Database#wr
 - [ ] Assigned two reviewers
 - [ ] Guarded against already existing resources using `IF NOT EXISTS` and `IF EXISTS`
 - [ ] Downs created to undo changes in Ups
+- [ ] Data migrations aren't being done (if so, a [Durable Job]((https://github.com/civiform/civiform/wiki/Database#durable-jobs-for-data-updates) should likely be used)
```
Comment (rockycodes) [inline .github/pull_request_template.md:30]: suggestion: this language is a little confusing, at first read I thought it was saying Durable Jobs should be used if data migrations aren't being done. What do you think of this:
`Data migrations aren't being done (please use a [Durable Job](link) if doing a data migration`

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -26,6 +27,14 @@ Read the guidelines [here](https://github.com/civiform/civiform/wiki/Database#wr
...
+
+#### Durable jobs
+
+Read the docs [here](https://github.com/civiform/civiform/wiki/Database#durable-jobs-for-data-updates)
+
+- [ ] Assigned two reviewers
+- [ ] Included a rollback plan
```
Comment (rockycodes) [inline .github/pull_request_template.md:37]: suggestion (non-blocking): `Included a rollback plan and a job to undo the data changes if appropriate.`

File: .github/pull_request_template.md (hunk 1 above)
Comment (dkatzz) [reply .github/pull_request_template.md]: Updated!

File: .github/pull_request_template.md (hunk 2 above)
Comment (dkatzz) [reply .github/pull_request_template.md]: Updated!

**Review by rockycodes — APPROVED** [review]

**Pull request merged**